x-pack/plugins/lens/public/shared_components/color_palette/color_ranges.tsx

```diff
@@ -0,0 +1,419 @@
+          defaultMessage: 'Add color range',
+        })}
+        size="xs"
+        flush="left"
+        disabled={shouldDisableAdd}
+        onClick={() => {
+          const newColorRanges = [...localColorRanges];
+          const length = newColorRanges.length;
+          const { max } = getDataMinMax(rangeType, dataBounds);
+          const step = getStepValue(
+            colorRanges.map(({ color, start }) => ({ color, stop: Number(start) })),
+            newColorRanges.map(({ color, start }) => ({ color, stop: Number(start) })),
+            max
+          );
+          const prevColor = localColorRanges[length - 1].color || DEFAULT_COLOR;
+          const newStart = step + Number(localColorRanges[length - 1].start);
+          const prevEndValue = newColorRanges[length - 1].end;
+          newColorRanges[length - 1].end = newStart;
+          newColorRanges.push({
+            color: prevColor,
+            start: newStart,
+            end:
+              newStart < prevEndValue
+                ? prevEndValue
+                : dataBounds.max > newStart
+                ? dataBounds.max
+                : newStart + step,
+          });
+          setColorRanges(newColorRanges);
+        }}
+      >
+        {i18n.translate('xpack.lens.dynamicColoring.customPalette.addColorRange', {
+          defaultMessage: 'Add color range',
+        })}
+      </EuiButtonEmpty>
+      <EuiButtonEmpty
+        data-test-subj={`lnsPalettePanel_dynamicColoring_reverseColors`}
+        iconType="sortable"
+        color="primary"
+        aria-label={i18n.translate('xpack.lens.dynamicColoring.customPalette.reverseColors', {
+          defaultMessage: 'Reverse colors',
+        })}
+        size="xs"
+        flush="left"
+        onClick={() => {
+          setColorRanges(reversePalette(localColorRanges));
+        }}
+      >
+        {i18n.translate('xpack.lens.dynamicColoring.customPalette.reverseColors', {
+          defaultMessage: 'Reverse colors',
+        })}
+      </EuiButtonEmpty>
+      <EuiButtonEmpty
+        data-test-subj={`lnsPalettePanel_dynamicColoring_distributeEqually`}
+        iconType={DistributeEquallyIcon}
+        color="primary"
+        aria-label={i18n.translate('xpack.lens.dynamicColoring.customPalette.distributeEqually', {
+          defaultMessage: 'Distribute equally',
+        })}
+        size="xs"
+        flush="left"
+        onClick={() => {
+          const colorsCount = localColorRanges.length;
+          const start = localColorRanges[0].start;
+          const end = localColorRanges[colorsCount - 1].end;
+          const step = roundValue((end - start) / colorsCount);
+          const newRanges = localColorRanges.map((colorRange, index) => {
+            return {
+              color: colorRange.color,
+              start: roundValue(start + (step * 100 * index) / 100),
+              end: roundValue(start + (step * 100 * (index + 1)) / 100),
```

Comment:
Also here I think the `* 100 / 100` are superfluous. The `roundValue` will apply already some internally

Reply:
no because firstly we should use `*100` and after that  `* index / 100` in another case we have problem. You can check it try with that expression `11596411699.2 * 3 * 100 / 100` and `11596411699.2 * 100 * 3/ 100` you will get different results

Reply:
But that's not a different result, it is just how float values work. But even when slightly different after the 5th digit, the `roundValue` will take care of it.

Reply:
sometimes I got `.59` instead of `.6` if i use the first options, with the second always `.6`.